### PR TITLE
Introduce 30% treshold for validator unstake

### PR DESCRIPTION
**Pull request by fabiankrol**
ref: `Limit validate withdrawal beyond a certain percentage` #3921

### CODE REVIEW

File: test/contracts/MainStaking.aes [hunk 1]
```diff
@@ -57,6 +59,7 @@ main contract MainStaking =
       entropy               = Crypto.sha256(entropy_str),
       leader                = Contract.address,
       validator_min_stake   = validator_min_stake,
+      validator_min_percent = 30,
```
Comment (fabiankrol) [inline test/contracts/MainStaking.aes:62]: TODO: hardcoded value should be configurable

Comment (fabiankrol) [reply test/contracts/MainStaking.aes]: ✅

File: test/contracts/MainStaking.aes [hunk 2]
```diff
@@ -237,14 +243,25 @@ main contract MainStaking =
...
       true
+
+  function assert_min_stake(balance : int) =
+      if(balance < state.validator_min_stake)
+        abort("Validator can not withdraw below the treshold")
+      else false
+
+  function assert_min_percent_stake(balance : int) =
+      let min_percent = state.validator_min_percent
+      if((100 * balance) < (min_percent * state.total_stake))
```
Comment (fabiankrol) [inline test/contracts/MainStaking.aes:265]: `state.total_stake` should not be used here

Comment (fabiankrol) [reply test/contracts/MainStaking.aes]: fixed

**Review by velzevur — APPROVED** [review]

**Pull request merged**